### ISSUE TITLE
question about valid

### ISSUE THREAD

**Issue by xuchen-dev** [ifnspaml/SGDepth#28]
i think train the model successful，i dont change anything in the code  the result looks good 
![image](https://user-images.githubusercontent.com/79962968/135188722-5c78446e-e96f-4a11-ad7d-d3b10132d27b.png)
it is about 10 epoch result

but the a1 a2 a3  rmse and so on is far from the paper privde
![image](https://user-images.githubusercontent.com/79962968/135189411-7a202755-d9e8-400d-85be-458548a19ad2.png)

is there something wrong in my depth gt 
i just use the code to get the gt depth from lidar point 
![image](https://user-images.githubusercontent.com/79962968/135188987-12e548a5-fadf-4276-b270-d7f44eaa1219.png)
![image](https://user-images.githubusercontent.com/79962968/135189020-6d65ed71-7f57-4ac0-8aad-809f8cee07e5.png)
the gt depth look like this 
![image](https://user-images.githubusercontent.com/79962968/135189129-cca3bdc7-51de-4bcb-aec1-137f4635e6d7.png)

is there something wrong in this way ，very hopeful for your helping   i am just a green hand students

**Comment by xuchen-dev** [comment]
![image](https://user-images.githubusercontent.com/79962968/135189539-b8f571a9-d748-48f6-be2c-34d252a6838e.png)
and my rmse

**Comment by xuchen-dev** [comment]
i download you provide sgddepth full trained model and eval it 
the result like this 
![1632883741(1)](https://user-images.githubusercontent.com/79962968/135194749-f185d3cd-be9e-41d0-8fcc-3a313eede8a2.png)
there must something wrong with my depth gt ， could you help me，i will be very grateful

**Comment by klingner** [comment]
Hi, 

up to now, I have never observed such an error. How did you prepare the data? Did the download_kitti.py script run successfully? Or did you prepare the data differently in any way?

**Comment by xuchen-dev** [comment]
thank for your replying ，i didnt use the  all the download_kitti.py， because i have downloaded it 
i prepare my data like this
 
![image](https://user-images.githubusercontent.com/79962968/135369922-6a6d8ae7-bf56-4039-8d80-46341e03ad87.png)

**Comment by xuchen-dev** [comment]
thank for your replying ，i didnt use the  all the download_kitti.py， because i have downloaded it 
i prepare my data like this
 
![image](https://user-images.githubusercontent.com/79962968/135369922-6a6d8ae7-bf56-4039-8d80-46341e03ad87.png)

**Comment by klingner** [comment]
Hi, so what I observe in the output of the depth evaluation is that your scaling ratio is rather unusual (0.023). I usuall have scaling ratios in the order of 20-30 (deviation of a factor of $10^3$), so I would agree that something regarding you depth gt is wrong. You can download the depth gt for the image ``kitti/Depth/2011_09_26/2011_09_26_drive_0001_sync/image_02/data/0000000000.png`` at this link:

https://drive.google.com/file/d/1QRudmxq5ENl4tfXuO2A_n381_VYRDtTX/view?usp=sharing

Maybe you can compare it to your depth gt and see, if they match up.

**Comment by xuchen-dev** [comment]
i wonder   the  ratio is   ratio = (median_gt / median_pred).item()     i wonder  the lower the ratio   the closer the value between  gt and pred  why normal vlaue is 20 - 30 ， i just use the generate_depth_from_velo  in  the kitti_download to get depth gt , i dont know i got this problem 
![image](https://user-images.githubusercontent.com/79962968/135742451-00ff5bae-5b68-4524-94e0-c644adb09a13.png)
![image](https://user-images.githubusercontent.com/79962968/135742473-c1e012a5-2855-48c9-a536-54280b1e6067.png)


yes t'here is something wrong in my depth gt ， i just use the generate_depth_from_velo  in  the kitti_download to get depth gt , i dont know  why i got this problem

**Comment by xuchen-dev** [comment]
and i wonder why your gt is png in the kitti_download code is
 
![image](https://user-images.githubusercontent.com/79962968/135742893-07d7683b-9e50-462b-b71f-07b82f76709a.png)
is it the way i get the depth gt is false ?

**Comment by klingner** [comment]
Hi, so I just set up the repository from scratch again and also tried out the download script. I can still exactly reproduce the results. What I also noted is that you store the images as jpg and not as png. In the unchanged download script the images are stored as png. I would recommend to try out the unchanged download script and then run the evaluation again.

**Comment by huhaoyue** [comment]
Hi, I want to run this program now, but I don't know how to arrange the data set. I don't know if I can get your guidance

**Comment by klingner** [comment]
Hi, to what extent did you follow the provided instructions in the README?

**Comment by huhaoyue** [comment]
> Hi, to what extent did you follow the provided instructions in the README?

I downloaded the dataset from the official website, and then followed the the latter part of download_kitti.py you gave to arrange the structure of the dataset, but the following error is reported when running tranin.py. 
![1640160339(1)](https://user-images.githubusercontent.com/82314113/147057602-ed7bbf42-e2ff-4550-ac62-9a43046fc189.png)

After debugging, I found that some of its samples are empty. Is this a problem with my dataset?
![1640160379(1)](https://user-images.githubusercontent.com/82314113/147057675-3a8931e8-fecc-4309-9720-574bae55afe4.png)

**Comment by huhaoyue** [comment]
> Hi, to what extent did you follow the provided instructions in the README?

![1640161588(1)](https://user-images.githubusercontent.com/82314113/147060594-a41456c4-8e20-48d5-acea-fd871c63ba94.png)
When I debug to this place, I can see that it is Nonetype. I don't quite understand where the problem is. If it is a dataset problem, how can I check where it appears?
![1640161617(1)](https://user-images.githubusercontent.com/82314113/147060677-d0a34272-abc8-4069-83b7-7178f1029005.png)

**Comment by klingner** [comment]
So for me it appears as if you are trying to load an image that does not exist at the specified path location, so likely the arrangement of your dataset is not yet correct. I would check in the file ``dataloader/pt_data_loader/basedataset.py`` in the function ``read_image_file(self, filepath)``, which filepath is loaded and if this path actually exist on your machine. Maybe this will give you a hint, if something has gone wrong when arranging the dataset.

**Comment by huhaoyue** [comment]
> 
This error occurs in the middle of the loop. Not all the data loaded is None, so I don't think it's a path problem. This is the amount of data I loaded. Is it correct?
![1640165639(1)](https://user-images.githubusercontent.com/82314113/147070612-dcacc8e6-f582-4df1-9099-86836199abe0.png)

**Comment by klingner** [comment]
Yes, the amount is correct, but it is not calculated from the data on disk but from the downloaded json-files, so I wouldn't expect this to be wrong. The None-error is typical for a non-existing image where the path is wrong.

**Comment by huhaoyue** [comment]
> 是的，这个数字是正确的，但它不是根据磁盘上的数据计算的，而是从下载的json文件中计算出来的，所以我不会认为这是错误的。对于路径错误的不存在的图像，"无"错误是典型的。

I checked that the image is missing here. Does the color means the RGB image corresponding to segmentation map？
![1640185407(1)](https://user-images.githubusercontent.com/82314113/147112704-cdff9906-5fec-4c99-b89a-55d059e1bcdb.png)

Does the None of this mean that the images captured by the camera in leftImg8bit corresponding to the segmentation maps in gtFine are not loaded?

The structure of my cityscapes dataset is as follows:
--data
   --cityscapes
        --gtFine
            --train
                --aachen
                --bochum
                --bremen
                --cologne
                      ....
             --test
             --val
         --leftImag8bit_trainvaltest
               --train
               --val
               --test
     --kitti
        --Depth
        --Raw_data
     --kitti_kitti_split
         --.json
     --kitti_zhou_split
         --.json

**Comment by huhaoyue** [comment]
> 

Thank you very much for your reply. I found the problem. The reason for the error is that I found above. As long as the name of the dataset is changed, just change the leftimage8bit_ trainvaltest in cityscapes to leftimage8bit. This program can run normally.
